### DISCUSSION TITLE
Automations and timers

### DISCUSSION THREAD

**Discussion by Crlaozwyn** [catgiggle/OctoPrint-GpioControl#17] [Ideas]
Fantastic GPIO plugin - thank you. Would you be willing to consider including timers or controls tied to prints? For example, when a print starts, automatically turn on an input controlling a light, then turn it off two minutes after the print finishes.

I saw that you're planning a more advanced GPIO control plugin in future, but figured I'd check just in case it makes sense here. Again, appreciate your efforts on this.

**Comment by catgiggle** [comment]
That's right, automatons will be part of a more advanced plugin. But good point with timers! It makes sense.

BTW. Thanks. I'm glad you like my plugin :)

**Reply by fank** [reply]
I don't see the reason for a more advanced plugin, i would like to have all gpio abilities in one plugin.
Its just an ui design choice of how easy or complex it will be.

I would like to have the current control setup, with an option to define triggers on the buttons or in an additional view.
The best example i have seen so far is the "UI Customizer" configuration options, its quite easy to handle but still so many options to adjust.

**Reply by catgiggle** [reply]
You will have all gpio abilities in one plugin. This "mysterious more advanced plugin" will have all GPIO Control features and much more.

I want to keep this plugin as simple as possible and I think that switching is enough for a lot of users. If it turns out that this more advanced version is chosen much more often and there is no need to maintain two plugins, I will give up one. It's less work for me :)

Thanks for idea. Could you list triggers that will be useful for you?

**Reply by fank** [reply]
So far from what i have seen is possible (somehow) and would be useful for me:

- Progress in percent
- Progress in layer (Maybe depends on https://github.com/OllisGit/OctoPrint-DisplayLayerProgress , because of stuff i only want to turn on after the first 10 layers)
- Time (remaning, starting at 30min or less i can turn off the heatbed and start the heatbed cooling)
- Print state change (started/(un)paused/stopped, only would like to have lights enabled as long is a print is running, because of webcam recording)
